featurebyte/feast/infra/offline_stores/spark_thrift_source.py

```diff
@@ -0,0 +1,267 @@
+            data_source_class_type="featurebyte.feast.infra.offline_stores.spark_thrift_source.SparkThriftSource",
+            field_mapping=self.field_mapping,
+            custom_options=self.spark_thrift_options.to_proto(),
+            description=self.description,
+            tags=self.tags,
+            owner=self.owner,
+        )
+
+        data_source_proto.timestamp_field = self.timestamp_field
+        data_source_proto.created_timestamp_column = self.created_timestamp_column
+
+        return data_source_proto
+
+    def validate(self, config: RepoConfig) -> None:
+        self.get_table_column_names_and_types(config)
+
+    @staticmethod
+    def source_datatype_to_feast_value_type() -> Callable[[str], ValueType]:
+        return spark_to_feast_value_type
+
+    def get_table_column_names_and_types(self, config: RepoConfig) -> Iterable[Tuple[str, str]]:
+        return cast(
+            Iterable[Tuple[str, str]],
+            config.offline_store.get_table_column_names_and_types(self.get_table_query_string()),
+        )
+
+    def get_table_query_string(self) -> str:
+        """
+        Returns a string that can directly be used to reference this table in SQL
+
+        Returns
+        -------
+        str
+        """
+        return f"`{self.catalog}`.`{self.schema}`.`{self.table}`"
+
+
+class SparkThriftOptions:
+    """
+    DataSource options for Spark Thrift Server
+    """
+
+    def __init__(
+        self,
+        catalog: str,
+        schema: str,
+        table: str,
+    ):
+        self.catalog = catalog
+        self.schema = schema
+        self.table = table
+
+    @classmethod
+    def from_proto(
+        cls, spark_thrift_options_proto: DataSourceProto.CustomSourceOptions
+    ) -> "SparkThriftOptions":
+        """
+        Creates a SparkThriftOptions from a protobuf representation of a SparkThriftOptions
+
+        Parameters
+        ----------
+        spark_thrift_options_proto: DataSourceProto.CustomSourceOptions
+            A protobuf representation of a SparkThriftOptions
+
+        Returns
+        -------
+        SparkThriftOptions
+        """
+        config = json.loads(spark_thrift_options_proto.configuration.decode("utf8"))
+        spark_thrift_options = cls(
+            catalog=config["catalog"], schema=config["schema"], table=config["table"]
+        )
+        return spark_thrift_options
+
+    def to_proto(self) -> DataSourceProto.CustomSourceOptions:
+        """
+        Converts a SparkThriftOptions to its protobuf representation
+
+        Returns
+        -------
+        DataSourceProto.CustomSourceOptions
+        """
+        spark_thrift_options_proto = DataSourceProto.CustomSourceOptions(
+            configuration=json.dumps(
+                {"catalog": self.catalog, "schema": self.schema, "table": self.table}
+            ).encode()
+        )
+        return spark_thrift_options_proto
+
+
+class SavedDatasetSparkThriftStorage(SavedDatasetStorage):
```

Comment:
What is the purpose of this class? Do we need the same for databricks?

Reply:
It does not seem to be used in our serving implementation - will remove.

We are using the same data source class (SparkThriftSource) for spark, databricks and databricks unity, the difference is only in the offline store configs where each offline store type uses a different implementation for establishing a thrift session